File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        vector<int> result(k);
        unordered_map<int, int> counts;

        for (int num : nums) {
            ++counts[num];
        }

        std::priority_queue<pair<int, int>> frequency;
        for (auto& [num, count] : counts) {
            frequency.push({count, num});
        }

        for (int i = 0; i < k; ++i) {
            result[i] = frequency.top().second;
            frequency.pop();
        }

        return result;
    }
};

File: Data Structures & Algorithms/baseball-game/submission-0.py
class Solution:
    def calPoints(self, operations: List[str]) -> int:
        score = 0
        result = [] 

        for op in operations:            
            if op == "+":
                result.append(result[-1] + result[-2])

            elif op == "D":
                result.append(result[-1] * 2)
            
            elif op == "C":
                result.pop()

            else:
                result.append(int(op))

        for r in result:
            score += int(r)

        return int(score)
